### ISSUE TITLE
Toxin GLA Demo Trap is not practical in regular match

### ISSUE THREAD

**Issue by ExiLe-ZH** [TheSuperHackers/GeneralsGamePatch#802]
Costs $600, builds very slow, ~~automatic detonation reacts slow,~~ very low damage output.

## Demo Trap Specs

Demo Trap | price | build time (seconds) | primary damage | 30 s anthrax beta damage | Prerequisite
-- | -- | -- | -- | -- | --
vGLA | 400 | 10 | 600 | - | Arms Dealer 
Tox | 600 | 16 | 250 | 140 | Arms Dealer 
Demo | 200 | 6 | 700 |  - | Supply Stash 
Stealth | 400 | 10 | 600 | - | Arms Dealer 

## Contamination Times

| Object                   | Contamination Time (s) |
|--------------------------|------------------------|
| Toxin Shells (Scorpion)  | 10                     |
| Toxin Tractor Sprinkler  | 180                    |
| SCUD Storm               | 30                     |
| SCUD Launcher Anthrax    | 30                     |
| Bio Bomb Truck           | 30                     |
| Demo Trap                | 30                     |
| Toxin Bomb               | 60                     |

## Puddle Damages

Demo Trap uses MediumPoisonFieldWeaponUpgraded and Chem_MediumPoisonFieldWeaponGamma.

| Weapon                                          | Damage | DPS  |
|-------------------------------------------------|:-------|:-----|
| SmallPoisonFieldWeapon (Toxin Shells)           | 2.0    | 3.75 |
| SmallPoisonFieldWeaponUpgraded (Toxin Shells)   | 2.5    | 4.69 |
| Chem_SmallPoisonFieldWeaponGamma (Toxin Shells) | 2.5    | 4.69 |
| MediumPoisonFieldWeapon                         | 2.0    | 3.75 |
| MediumPoisonFieldWeaponUpgraded                 | 2.5    | 4.69 |
| Chem_MediumPoisonFieldWeaponGamma               | 2.5    | 4.69 |
| LargePoisonFieldWeapon (Anthrax Bomb)           | 15     | 28.1 |
| LargePoisonFieldWeaponUpgraded (Anthrax Bomb)   | 25     | 46.9 |
| Chem_LargePoisonFieldWeaponGamma (Anthrax Bomb) | 25     | 46.9 |
| AnthraxBombPoisonFieldWeapon                    | 40     | 75.0 |
| AnthraxBetaBombPoisonFieldWeapon                | 40     | 75.0 |
| AnthraxGammaBombPoisonFieldWeapon               | 40     | 75.0 |
| SmallRadiationFieldWeapon (Unit death puddles)  | 5      | 6.5  |
| Nuke_SmallRadiationFieldWeapon (Battle Master)  | 15     | 19.6 |
| MediumRadiationFieldWeapon                      | 15     | 19.6 |
| LargeRadiationFieldWeapon                       | 25     | 32.6 |
| NukeRadiationFieldWeapon (Nuke Missile)         | 25     | 32.6 |

### Proposal 1
Decrease price

### Proposal 2
Increase build speed (Done)
* #2015

### ~~Proposal 3~~
~~Improve reaction time~~

### Proposal 4
Buff damage output

### Proposal 5
Increase explosion damage

### Proposal 6
Increase toxin puddle duration

### Proposal 7
Change prequisite to unlock it earlier in the tech tree. 

### Proposal 8
Increase toxin puddle radius

### Proposal 9
Increase Anthrax Gamma puddle damage (Done)
* #2012

**Comment by ExiLe-ZH** [comment]
Personally I don't think the Tox Demo Trap needs any improvement strength wise vs vehicles (except maybe Techs), being bit more effective vs infantry couldn't hurt I suppose.

Overall my preference goes to tweaking Build Time and Price, since Tox in 1.04 doesn't need any improvements from balance perspective.

**Comment by commy2** [comment]
| Demo Trap | price | build time (seconds) | primary damage | 30 s anthrax beta damage |
|-----------|-------|------------|----------------|-----------|
| vGLA      | 400   | 10         | 600            |           |
| Tox       | 600   | 16         | 250            | 140       |
| Demo      | 200   | 6          | 700            |           |
| Stealth   | 400   | 10         | 600            |           |

**Comment by Jundiyy** [comment]
It is already very good vs Infantry. They will die in the pool of toxins.
I would just like to reduce BT and price.

**Comment by commy2** [comment]
> It is already very good vs Infantry.

They are all equally good against infantry, because the initial blast kills them regardless of any poison.

**Comment by ExiLe-ZH** [comment]
Well actually better reaction time and/or cloud radius would be nice vs tech terrors

**Comment by Jundiyy** [comment]
> > It is already very good vs Infantry.
> 
> They are all equally good against infantry, because the initial blast kills them regardless of any poison.

Yes true but with Toxin one, if there are a few infantry, one sets it off, the rest will die in the toxins.
Even if a hero is out of the original radius and it is set off, the toxins would finish off or lower the hp.

**Comment by RisingZH** [comment]
I would leave it as it is. Tox has many things which are superior to the other glas, makes sense that some things are inferior. 

It is good for factions to have unique aspects.

**Comment by xezon** [comment]
To me price and build time looks unreasonable for something that performs worse. As for it causing less damage, that is probably ok. I estimate 400 cost and 10 or 12 seconds build time would be acceptable - reason being, 600 cost does not warrant building this. Player can spend 200 money more and build a Toxin Tunnel for much more practical value in return. And less build time because it can make a huge difference to being it useful vs wasted. If enemy can see for 16 second that a Worker swings his hammer in the air, then it is much less likely for the Toxin Trap to succeed in a surprise.

**Comment by ReLaX82** [comment]
Increase radius and reduce cost to 300 or 400.

**Comment by kABUSE1** [comment]
> Well actually better reaction time and/or cloud radius would be nice vs tech terrors

I don't understand why you say that their reaction time is worse over other demo traps, as it's the polar opposite. They blow up DIRECTLY once triggered (for both auto and manual mode). Maybe your manual timing is off because you are not used to this?

But yes, it is worth discussing whether that's a bug or not. I am not 100% sure what is worse either.

Tox demo traps are more effective against very fast units like technicals or rocket buggies, but might be too quick for other unit types. A ridiculously complex way of fixing this while keeping all use-cases covered would be to have 3 modes on all demo trap types. "manual mode, direct mode, delay mode". Then you would still have to fix the "manual mode" so it behaves the same for all types of traps. This could be very interesting as it would add an entirely new meta to demo-trap handling.

Worth to keep in mind regarding demo trap balancing is that demo general traps can be build with only a supply stash, while all other types require their respective arms dealer.

**Comment by xezon** [comment]
The missing delay of the Toxin Trap was fixed with #85.

Does this mean the Toxin Trap is now weaker in Patch?

**Comment by MTKing4** [comment]
Not necessarily, delay is useful against a clump of units, giving enough time for more units to get closer to get maximum damage possible

**Comment by xezon** [comment]
The Anthrax Beta Toxin field deals 2.5 / 0.533 * 30 = 140 POISON damage. POISON damage is weak against vehicles.

**Comment by xezon** [comment]
I do wonder if Tox Trap has 16 seconds build time because developer forgot to type build time in halved time: 4 seconds instead of 8 seconds. Perhaps the intent was 4 (*2) seconds really.

**Comment by xezon** [comment]
To make Toxin Trap better vs Infantry, we could make its poison puddle stay longer, 60 seconds instead of 30 seconds. Or increase toxin puddle radius, as already proposed.

**Comment by ExiLe-ZH** [comment]
I would be fan of making it better vs infantry (and even less effective vs vehicles if necesarry) and equalizing price and build time with other traps. It sounds pretty safe to me https://github.com/TheSuperHackers/GeneralsGamePatch/pull/2015#issuecomment-1596084552.

**Comment by xezon** [comment]
I do wonder though, if its utility is primarily against infantry, does it make it more useful then in practice?

It is rather unusual that blobs of infantry units attack - apart from Heros and Angry Mobs.

**Comment by ExiLe-ZH** [comment]
Perhaps we can change the Tox Trap's prerequisite aswell, give it a little advantage by making it available after the Barracks, given it's an anti-infantry weapon. ~~The slow reaction time also suggests this, as infantry moves slow.~~ 

Tbf though, the whole anti-inf thing, it's very niche. In practice nobody sends inf blobs to fight Tox. I guess mostly useful to catch heroes or indeed mobs.

**Comment by xezon** [comment]
The reaction time of Toxin Trap is 1000 ms, same as all other Traps.

**Comment by ExiLe-ZH** [comment]
Really? I guess a case of placebo then. 

Is Demo also 1000 ms?

**Comment by xezon** [comment]
Yes. All 1000 ms. In Patch.

**Comment by xezon** [comment]
Barracks Prerequisite is interesting idea. The Demo Trap Prerequisite is Demo_GLASupplyStash.

**Comment by ExiLe-ZH** [comment]
Added prerequisite to table

**Comment by xezon** [comment]
Toxin Trap build time was decreased from 16 to 8 seconds. This makes it more attainable and useful. From my side this task is done. Up to you if you want anything else done for it.

Note that easing poison and radiation puddle stacking behavior could make the Toxin Trap better.

* #55

**Comment by xezon** [comment]
Contamination times added to opening post because relevant.

**Comment by ExiLe-ZH** [comment]
Would be good to increase it, atleast 60 seconds, maybe even same like Toxin Tractor. 

Would do the same with Tox Scud Launcher and Bomb Truck, since they both suck aswell.

**Comment by xezon** [comment]
180 seconds (3 minutes) is too long in my opinion for any of these.

I also added Toxin and Radiation DPS information in opening post, because relevant. Small and Medium Radiation are about 4 times more potent than its Toxins counterparts. On the other hand the Anthrax Bomb is 1.44 times more potent than the Nuke Missile Radiation.

**Comment by ExiLe-ZH** [comment]
The idea is area denial, basically putting up chemical minefields by manually detonating these traps, I always imagined Toxin could do such thing.

**Comment by xezon** [comment]
Proposal 9 done.
